hardzeichyksiarhei/shop-nodejs-serverless
# Task9/Backend For Frontend

1. https://github.com/rolling-scopes-school/nodejs-aws-tasks/blob/main/task9-bff/task.md
2. 26.10.2021 / 06.11.2021
3. Deploy: - (local run & remote link)
4. Done 05.11.2021 / deadline  06.11.2021
5. Score: 7 / 7

---


**Link to BFF (root)** – http://hardz-bff-api-dev.eu-west-3.elasticbeanstalk.com
**Link to BFF (products)** – http://hardz-bff-api-dev.eu-west-3.elasticbeanstalk.com/products
**Link to BFF (cart)** – http://hardz-bff-api-dev.eu-west-3.elasticbeanstalk.com/cart


---


## Required functionality:

- [x] +2 - The bff-service should be deployed with Elastic Beanstalk. The bff-service call should be redirected to the appropriate service : product-service or CART. The response from the bff-service should be the same as if product-service or CART services were called directly.
![image](https://user-images.githubusercontent.com/11542402/140619217-9a24e1b4-c9a3-465f-a551-bf6df74bc069.png)

## Additional tasks:

      **How to test:**
      - Get products list
      - Create new product
      - Get products list - result shouldn’t have new product
      - Wait more than 2 minutes
      - Get products list - result should have new product


---


## ❗❗❗ Troubleshooting ❗❗❗
Mixed Content: The page at '{url}' was loaded over HTTPS, but requested an insecure XMLHttpRequest

**Problem:**
![image](https://user-images.githubusercontent.com/11542402/140619065-b88a05e3-77ca-4134-bc8b-cddbef844b52.png)

**Solution:**
![image](https://user-images.githubusercontent.com/11542402/138672697-d7a32436-e37c-439e-8715-dd0b2650476b.png)
![image](https://user-images.githubusercontent.com/11542402/138672810-8d2aa458-5009-45e3-8255-21e2e920a0cf.png)
![image](https://user-images.githubusercontent.com/11542402/140619093-419f41cd-a2a9-403a-87d9-518f62ec9bbf.png)


---


## Simple examples:
![image](https://user-images.githubusercontent.com/11542402/140619519-7c26606c-f76d-4fd2-b0e8-35af458948b3.png)
![image](https://user-images.githubusercontent.com/11542402/140619460-1ece8968-2fbd-49ce-baa8-4d3d6614a1bc.png)


Link to CloudFront deploy - https://d14waaxoxwy6d2.cloudfront.net

Link to FE MR - https://github.com/hardzeichyksiarhei/shop-vue-vuex-cloudfront/pull/8

## Total: 7
